src/components/inputField/InputField.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState } from "react";
+import InputFieldDiv from "../../types/InputField";
+import SelectElement from "./selectElement/SelectElement";
+import SearchIcon from "../../assets/images/icon-search.svg?react";
+
+import './inputField.scss';
+
+export default function InputField({selectElement, inputType, inputDecorator, inputPlaceholder, selectDecorator, fieldTitle, helperText, onChange} : InputFieldDiv) {
+    const [dropdownOpened, setDropdownOpened] = useState(false); 
+
+    function toggleDropdown(){
+        setDropdownOpened(!dropdownOpened); 
+    }
+    function closeDropdown(){
+        setDropdownOpened(false)
+    }
+
+    return(
+        <div className="input-field">
+            {/* Field title if it exists */}
+            {fieldTitle ? 
+            <p className="field-title">{fieldTitle}</p>
+            : null}
+
+            {/* Either input or select element +/- decorators */}
+            <div className="input-box" 
+                tabIndex={0}
+                onClick={toggleDropdown}
+                onBlur={closeDropdown}
+            >
+                {selectElement ?
+                    <SelectElement dropdownOpened={dropdownOpened} selectDecorator={selectDecorator}
```

Comment:
I think the SelectElement could be used as a separated element instead of including it in the Input. This way, both components can be simplified. 
Also, if you want a way to reuse the field title and the input decorators and so on, you can use the principle of composition (aka using children).

You would have a InputShellComponent where you put what's before and after the actual input, and an InputBodyComponent and the SelectElementComponent.

The you can create the main usable components for the Input and Select like this

```
const InputComponent = (props) => <InputShellComponent {...props}><InputBody {...props}/></InputShellComponent>

const SelectComponent = (props) => <InputShellComponent {...props}><SelectElement {...props}/></InputShellComponent>
```

This is a rough sketch of the way you can achieve this with composability, maybe the props spreading is not the best thing to do here.